### ISSUE TITLE
Return specific error values for Insert(...).RunWrite(...)

### ISSUE THREAD

**Issue by nstogner** [rethinkdb/rethinkdb-go#274]
As far as I can tell the only way of determining **why** a call to RunWrite has failed is by parsing the error string.

For instance if I want to return a 409-Conflict in my REST api when an insert fails due to an existing document, as opposed to a 500-Internal-Server-Error for other error cases, I need to parse the error string for the text: "Duplicate primary key". This seems brittle.

Could a specific error value be returned (ErrConflict) for such cases?

**Comment by nstogner** [comment]
Any thoughts?

**Comment by dancannon** [comment]
Hey, really sorry for taking so long to reply. I have been struggling to spend time working on GoRethink over the last couple of weeks. Hopefully that will change soon.

Regarding your question the "Duplicate primary key" error comes directly from RethinkDB and is just the `FirstError` field extracted from the `WriteResponse`. To return a specific error GoRethink would also have to parse the error message however I am not too keen to change the API by returning a different error message (especially one which has less information than the current error which some users might be relying on). 

An alternative solution would be to provide some helper functions for checking error type (such as `IsConflictErr()`). Do you think this would help you out?

**Comment by nstogner** [comment]
Thanks for your response and all the work you have done on this library. So if I understand you, RethinkDB does not return specific error codes for why an insert failed? It just spits out a string that 3rd party drivers need to parse?

I can not decide if a helper function which essentially parses a string (that may change in future RethinkDB releases?) would be better than having users of GoRethink do the parsing themselves.

**Comment by dancannon** [comment]
Thanks! You are correct, RethinkDB returns the following JSON object without any more information. As you can see there is not much else that can be done other than doing something like `strings.Contains`. I think I will probably add some helper functions as it might make working with these errors easier but you are right it is not much better.

``` json
{"deleted":0,"errors":1,"first_error":"Duplicate primary key `id`:\n{\n\t\"id\":\t\"1\"\n}\n{\n\t\"id\":\t\"1\"\n}","inserted":0,"replaced":0,"skipped":0,"unchanged":0}
```

**Comment by dancannon** [comment]
I have included the helper functions in 1.4.0, I know this isnt ideal but hopefully it helps a bit.

**Comment by nstogner** [comment]
Thanks for your work!
